tests/test_env.py

```diff
@@ -288,6 +288,36 @@ def test_process_env_install_from_index_not_found_env_var(
         )
 
 
+def test_process_env_install_no_index(modules_v2_dir) -> None:
+    """
+    Attempt to install a package that does not exist with --no-index.
+    To have --no-index set in the pip cmd, the config should not contain an index_url,
+    we should not be using the system config and a path needs to be specified.
+    Assert the appropriate error is raised.
+    """
+
+    # Setup a temporary directory as project_dir
+    with tempfile.TemporaryDirectory() as temp_dir:
+        project_dir = temp_dir
+        setup_py_content = """
+from setuptools import setup
+setup()
```

Comment:
A Python package without a name feels a bit weird to me.
```suggestion
setup(name="test")
```